### ISSUE TITLE
Support other architectures (armhf)

### ISSUE THREAD

**Issue by explorigin** [SteveLTN/https-portal#152]
This would be great for a RaspberryPi, but currently *amd64* is hardcoded in several places.  Please at least parameterize the architecture.

**Comment by explorigin** [comment]
...please.  😄

**Comment by SteveLTN** [comment]
Hi,
I would love to do that, but unfortunately I don't have a device to test it out. If you feel like to, please submit a pull-request. Shouldn't be too hard, just need to replace the `amd64` in `Dockerfile` with a parameter I think. The base image `nginx` does support armhf.

**Comment by explorigin** [comment]
I was able to get this running on my RPi3b+ by using this dockerfile:

```dockerfile
FROM nginx:1.12.2

MAINTAINER Weiyan Shao "lighteningman@gmail.com"

WORKDIR /root

ENV DOCKER_GEN_VERSION 0.7.3
ENV ARCH armhf

ADD https://github.com/just-containers/s6-overlay/releases/download/v1.18.1.0/s6-overlay-$ARCH.tar.gz /tmp/
ADD https://github.com/jwilder/docker-gen/releases/download/$DOCKER_GEN_VERSION/docker-gen-linux-$ARCH-$DOCKER_GEN_VERSION.tar.gz /tmp/
ADD https://raw.githubusercontent.com/diafygi/acme-tiny/ad7802f1c47e5c31a8e7dfedb3577e6c7d04844a/acme_tiny.py /bin/acme_tiny

RUN tar xzf /tmp/s6-overlay-$ARCH.tar.gz -C / &&\
    tar -C /bin -xzf /tmp/docker-gen-linux-$ARCH-$DOCKER_GEN_VERSION.tar.gz && \
    rm /tmp/docker-gen-linux-$ARCH-$DOCKER_GEN_VERSION.tar.gz && \
    rm /tmp/s6-overlay-$ARCH.tar.gz && \
    rm /etc/nginx/conf.d/default.conf && \
    apt-get update && \
    apt-get install -y python ruby cron iproute2 apache2-utils && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*

COPY ./fs_overlay /

RUN chmod a+x /bin/*

VOLUME /var/lib/https-portal

ENTRYPOINT ["/init"]
```

I don't have a setup for cross building right now.  But other interested persons can copy-pasta the above and have a working setup. 

2 things are noteworthy:
- I had to update s6-overlay
- Generating the DHparam on an RPi takes hours.  There is no way around this (precomputing this defeats its purpose).  Just thought if anyone wants this on their home server, be ready to wait on the first run.

**Comment by inductor** [comment]
https://github.com/SteveLTN/https-portal/pull/201

I'm working on upgrading s6-overlay.

**Comment by inductor** [comment]
It’s merged now! @explorigin :)

**Comment by SteveLTN** [comment]
And I released 1.7.2.
